examples/README.rst

```diff
@@ -10,22 +10,87 @@ Building Notebooks
+Since some of the examples require a considerable amount of memory (``deconv_microscopy_tv_admm.py`` and ``deconv_microscopy_allchn_tv_admm.py`` in particular), it is recommended to set the following environment variables prior to building the notebooks:
 
-2. Run ``makeindex.py`` to update the example scripts README file, the notebook index file, and the examples index in the docs.
+::
+
+  export XLA_PYTHON_CLIENT_ALLOCATOR=platform
+  export XLA_PYTHON_CLIENT_PREALLOCATE=false
+
+
+Running on a CPU
+^^^^^^^^^^^^^^^^
+
+If a GPU is not available, or if the available GPU does not have sufficient memory to build the notebooks, set the environment variable
+
+::
+
+  JAX_PLATFORM_NAME=cpu
+
+to run on the CPU instead.
+
+
+Building Specific Examples
+--------------------------
+
+To build or rebuild notebooks for specific examples, the example script names can be specified on the command line, e.g.
+
+::
+
+  python makenotebooks.py ct_astra_pcg.py ct_astra_tv_admm.py
+
+When rebuilding notebooks for examples that themselves make use of ``ray``
+for parallelization (e.g. ``deconv_microscopy_allchn_tv_admm.py``), it is recommended to specify serial notebook execution, as in
+
+::
+
+  python makenotebooks.py --no-ray deconv_microscopy_allchn_tv_admm.py
 
-3. Run ``makenotebooks.py`` to build the new notebook, as well as any other notebooks that are out of date with respect to their source scripts, as determined by the respective file timestamps.
 
-4. Add and commit the new script, the ``scripts/index.rst`` script index file, the auto-generated ``scripts/README.rst`` file and ``docs/source/examples.rst`` index file, and the new or updated notebooks and the auto-generated ``notebooks/index.ipynb`` file in the notebooks directory (following the submodule handling procedure as described in the developer docs).
+Building All Examples
+---------------------
 
+By default, ``makenotebooks.py`` only rebuilds notebooks that are out of date with respect to their corresponding example scripts, as determined by their respective file timestamps. However, timestamps for files retrieved from version control may not be meaningful for this purpose. To rebuild all examples, the following commands (assuming that GPUs are available) are recommended:
 
-The procedure for rebuilding notebook(s) after the source file(s) have been modified is:
+::
+
+  export XLA_PYTHON_CLIENT_ALLOCATOR=platform
+  export XLA_PYTHON_CLIENT_PREALLOCATE=false
+
+  touch scripts/*.py
+
+  python makenotebooks.py --no-ray deconv_microscopy_tv_admm.py deconv_microscopy_allchn_tv_admm.py
+
+  python makenotebooks.py
+
+
+Updating Notebooks in the Repo
+------------------------------
+
+The recommended procedure for rebuilding notebooks for inclusion in the ``data`` submodule is:
+
+1. Add and commit the modified script(s).
+
+2. Rebuild the notebook(s) as described above.
+
+2. Add and commit the updated notebooks following the submodule handling procedure described in the developer docs.
```

Comment:
'notebook(s)' for consistency

I would change all 'notebook(s)' to 'notebooks' though as optional singular is obvious.

Reply:
Agreed and fixed.